cpp/src/arrow/table.cc

```diff
@@ -32,8 +33,25 @@
 #include "arrow/util/stl.h"
 
 namespace arrow {
+namespace {
+using ::arrow::internal::checked_cast;
```

Comment:
I'm not sure why you moved this.

Reply:
minor code hygiene - do not introduce names to a namespace you don't own (here, the global namespace).